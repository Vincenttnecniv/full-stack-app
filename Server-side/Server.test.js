import renderer from "react-test-renderer";
import request from "supertest";
const Server = require("./Server");
//Heres where I run my tests for server.
describe("Server component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Server />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should return a 404 when invalid route", (done) => {
    request.post("Cannot GET /").expect(404).end(done);
  });
});
