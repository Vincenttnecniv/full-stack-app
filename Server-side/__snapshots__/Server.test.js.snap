// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server component matches the snapshot 1`] = `const express = require("express");
const app = express();
const port = 3000;
const cors = require("cors");
app.use(cors());

app.get("/music/:query/:media", (req, res) => {
  const request = require("request");
  let query = req.params.query;
  const media = req.params.media;
  const options = {
    method: "GET",
    url: `https://itunes.apple.com/search?term=${query}&media=${media}`,
    headers: {},
  };
  request(options, function (error, response) {
    if (error) throw new Error(error);
    res.send(response.body);
  });
});

app.delete("/delete/:id", (req, res) => {
  //Heres where I delete an project spesified by index user chooses
  let input = parseInt(req.params.id);
  const removeItem = { id: input };

  objects.splice(
    objects.findIndex((a) => a.id === removeItem.id),
    1
  );

  res.send(objects);
});

app.listen(port, () => {
  //The port I run on is http://localhost:3000/
  console.log(`Example app listening on port ${port}`);
});

`;
